mmv1/products/dlp/StoredInfoType.yaml

```diff
@@ -81,6 +89,15 @@ properties:
     name: 'displayName'
     description: |
       User set display name of the info type.
+  - !ruby/object:Api::Type::String
+    name: 'storedInfoTypeId'
+    description: |
+      The storedInfoType ID can contain uppercase and lowercase letters, numbers, and hyphens;
+      that is, it must match the regular expression: [a-zA-Z\d-_]+. The maximum length is 100
+      characters. Can be empty to allow the system to generate one.
+    immutable: true
+    default_from_api: true
```

Comment:
i'm a little confused how this can be `default_from_api` and also `url_param_only`
the `url_param_only` means it won't be returned in the api body, which aligns with my understanding of this field, but if that's the case there shouldn't be any `default_from_api` that we'll need to worry about. have you seen that this is returned from the api?

Reply:
Here, I've made the `stored_info_type_id` fields as `url_param_only` to avoid generating the flattener and expander functions for the same. We need to skip the expand function because the existing resource wraps the fields in terraform using the encoder `resourceDataLossPreventionStoredInfoTypeEncoder`. `storedInfoTypeId` is in the same hierarchy as this [wrapped object](https://cloud.google.com/dlp/docs/reference/rest/v2/projects.storedInfoTypes/create#request-body). Similarly, we need to skip the flatten function as the [GET endpoint](https://cloud.google.com/dlp/docs/reference/rest/v2/projects.locations.storedInfoTypes#resource:-storedinfotype) doesn't contain the `storedInfoTypeId` field as is; instead, we need to extract it from the `name` output-only field. Moreover, we want to make this field optional and computed. Hence, I've used `default_from_api`.

Reply:
Ah! ok, thanks!